pipeline/ingestion/src/main/java/org/datacommons/ingestion/data/GraphReader.java

```diff
@@ -0,0 +1,121 @@
+package org.datacommons.ingestion.data;
+
+import com.google.cloud.spanner.Mutation;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.datacommons.ingestion.spanner.SpannerClient;
+import org.datacommons.proto.Mcf.McfGraph;
+import org.datacommons.proto.Mcf.McfGraph.PropertyValues;
+import org.datacommons.proto.Mcf.McfGraph.TypedValue;
+import org.datacommons.proto.Mcf.McfOptimizedGraph;
+import org.datacommons.proto.Mcf.McfStatVarObsSeries.StatVarObs;
+import org.datacommons.proto.Storage.Observations;
+import org.datacommons.util.GraphUtils;
+
+public class GraphReader implements Serializable {
+  public static List<Node> graphToNodes(McfGraph graph) {
+    List<Node> nodes = new ArrayList<>();
+    for (Map.Entry<String, PropertyValues> entry : graph.getNodesMap().entrySet()) {
+      PropertyValues pvs = entry.getValue();
+      if (!GraphUtils.isObservation(pvs)) {
+        Map<String, McfGraph.Values> pv = pvs.getPvsMap();
+        Node.Builder node = Node.builder();
+        node.subjectId(entry.getKey());
+        node.name(GraphUtils.getPropertyValue(pv, "name"));
+        node.types(GraphUtils.getPropertyValues(pv, "typeOf"));
+        nodes.add(node.build());
+      }
+    }
+    return nodes;
+  }
+
+  public static List<Edge> graphToEdges(McfGraph graph) {
+    List<Edge> edges = new ArrayList<>();
+    for (Map.Entry<String, PropertyValues> nodeEntry : graph.getNodesMap().entrySet()) {
+      PropertyValues pvs = nodeEntry.getValue();
+      if (!GraphUtils.isObservation(pvs)) {
+        Map<String, McfGraph.Values> pv = pvs.getPvsMap();
+        String subjectId = nodeEntry.getKey(); // Use the map key as the subjectId
+        for (Map.Entry<String, McfGraph.Values> entry : pv.entrySet()) { // Iterate over properties
+          for (TypedValue val : entry.getValue().getTypedValuesList()) {
+            Edge.Builder edge = Edge.builder();
+            edge.subjectId(subjectId);
+            edge.predicate(entry.getKey());
+            edge.objectId(val.getValue());
```

Comment:
We should make sure to distinguish between object id (reference) and value (leaf node)